azurelinuxagent/common/rdma.py

```diff
@@ -199,10 +199,24 @@ def start(self):
     def process(self):
         try:
             RDMADeviceHandler.update_dat_conf(dapl_config_paths, self.ipv4_addr)
-            RDMADeviceHandler.wait_rdma_device(
-                self.rdma_dev, self.device_check_timeout_sec, self.device_check_interval_sec)
-            RDMADeviceHandler.write_rdma_config_to_device(
-                self.rdma_dev, self.ipv4_addr, self.mac_addr)
+
+            skip_rdma_device = False;
+            retcode,out = shellutil.run_get_output("modinfo hv_network_direct")
+            if retcode == 0 :
+                version = re.search("version:\s+(\d+)\.(\d+)\.(\d+)\D", out, re.IGNORECASE)
+                if version :
+                    v1 = version.groups(0)[0]
+                    v2 = version.groups(0)[1]
+                    if int(v1+v2)>40 :
```

Comment:
if you're trying to check v>4.1 this is not a good way to do it. v0.101 will pass this check too